Adyen/Core/Core Protocols/PaymentComponent.swift

```diff
@@ -28,13 +28,23 @@ public protocol PaymentComponent: PaymentAwareComponent, PaymentMethodAware {
 extension PaymentComponent {
     
     public func submit(data: PaymentComponentData, component: PaymentComponent? = nil) {
-        guard data.browserInfo == nil else {
-            delegate?.didSubmit(data, from: component ?? self)
+        let component = component ?? self
+        let checkoutAttemptId = component.context.analyticsProvider.checkoutAttemptId
+        let updatedData = PaymentComponentData(paymentMethodDetails: data.paymentMethod,
+                                               amount: data.amount,
+                                               order: data.order,
+                                               storePaymentMethod: data.storePaymentMethod,
+                                               browserInfo: data.browserInfo,
+                                               checkoutAttemptId: checkoutAttemptId,
+                                               installments: data.installments)
```

Comment:
here you can probably check if `data.checkoutAttemptId` is already equal to `component.context.analyticsProvider.checkoutAttemptId` before updating.

Reply:
Updated